tests/python/test_household.py

```diff
@@ -0,0 +1,377 @@
+import pytest
+import json
+from unittest.mock import MagicMock, patch
+from sqlalchemy.engine.row import LegacyRow
+
+from policyengine_api.routes.household_routes import household_bp
+from policyengine_api.services.household_service import HouseholdService
+
+from tests.fixtures.household_fixtures import (
+    SAMPLE_HOUSEHOLD_DATA,
+    SAMPLE_DB_ROW,
+    mock_database,
+    mock_hash_object,
+)
+
+
+class TestGetHousehold:
+    def test_get_existing_household(self, rest_client, mock_database):
+        """Test getting an existing household."""
+        # Mock database response
+        mock_row = MagicMock(spec=LegacyRow)
+        mock_row.__getitem__.side_effect = lambda x: SAMPLE_DB_ROW[x]
+        mock_row.keys.return_value = SAMPLE_DB_ROW.keys()
+        mock_database.query().fetchone.return_value = mock_row
+
+        # Make request
+        response = rest_client.get("/us/household/1")
+        data = json.loads(response.data)
+
+        assert response.status_code == 200
+        assert data["status"] == "ok"
+        assert (
+            data["result"]["household_json"] == SAMPLE_HOUSEHOLD_DATA["data"]
+        )
+
+    def test_get_nonexistent_household(self, rest_client, mock_database):
+        """Test getting a non-existent household."""
+        mock_database.query().fetchone.return_value = None
+
+        response = rest_client.get("/us/household/999")
+        data = json.loads(response.data)
+
+        assert response.status_code == 404
+        assert data["status"] == "error"
+        assert "not found" in data["message"]
+
+    def test_get_household_invalid_id(self, rest_client):
+        """Test getting a household with invalid ID."""
+        response = rest_client.get("/us/household/invalid")
+
+        assert response.status_code == 400
+        assert b"Invalid household ID" in response.data
+
+
+class TestCreateHousehold:
+    def test_create_household_success(
+        self, rest_client, mock_database, mock_hash_object
+    ):
+        """Test successfully creating a new household."""
+        # Mock database responses
+        mock_row = MagicMock(spec=LegacyRow)
+        mock_row.__getitem__.side_effect = lambda x: {"id": 1}[x]
+        mock_database.query().fetchone.return_value = mock_row
```

Comment:
**nitt/nonblocking** I tend to prefer a test support method (I would usually call this a fixture, but that may have specific technical implications here) like ``given_any_query_will_return(...)`` instead of mock-object-ese in the test. It makes for a cleaner read and if you change the way we query the DB you just need to change the function once, not every build test we wrote.

Reply:
I would love to chat with you about that. I'm new to the world of mock objects and fixtures, frankly, so curious to hear more during our meeting Wednesday.

Reply:
For now, I'm going to leave these as-is, but here's the pathway I'm thinking:

Finalize refactoring this and re-request review -> create "tactical PR" for testing that refactors this -> separately, address `policy` endpoint PR